modules/exploits/multi/kubernetes/exec.rb

```diff
@@ -170,11 +171,19 @@ def create_pod
       print_good("Pod created: #{pod_name}")
 
       print_status('Waiting for the pod to be ready...')
-      10.times do
-        sleep 3
+      ready = retry_until_true(timeout: datastore['PodReadyTimeout']) do
         pod = @kubernetes_client.get_pod(pod_name, namespace)
-        ready = pod.dig(:status, :containerStatuses).find { |status| status[:name] = @pod_name }[:ready]
-        break if ready
+        pod_status = pod[:status]
+        next if pod_status == 'Failure'
+
+        container_statuses = pod_status[:containerStatuses]
+        next unless container_statuses
```

Comment:
When you query too fast after attempting to create a pod, the `containerStatuses` element isn't present:

```
{
  "kind": "Pod",
  "apiVersion": "v1",
  "metadata": {
    "etc": "etc"
  },
  "spec": {
    "etc": "etc"
  },
  "status": {
    "phase": "Pending",
    "conditions": [
      {
        "type": "PodScheduled",
        "status": "True",
        "lastProbeTime": null,
        "lastTransitionTime": "2021-11-05T00:31:43Z"
      }
    ],
    "qosClass": "BestEffort"
  }
}
```